Chat App/View/Pub/Register/PubRegisterViewController.swift

```diff
@@ -0,0 +1,194 @@
+        viewModel = PubRegisterViewModel()
+        setupTextChangedHandlers()
+        setupObservers()
+    }
+    
+    override func loadView() {
+        super.loadView()
+        view = UIView()
+        view.backgroundColor = .systemGray3
+        createSubViews()
+    }
+    
+    @objc func registerButtonTapped() {
+        self.viewModel.inputs.registerUser(submittedUsername: usernameField.text!, password: passwordField.text!)
+    }
+    
+    @objc func loginButtonTapped() {
+        let loginViewController = PubChatLoginViewController()
+        self.navigationController?.pushViewController(loginViewController, animated: true)
+    }
+    
+    func setupTextChangedHandlers() {
+        let usernameValid = usernameField
+            .rx
+            .text
+            .observe(on: MainScheduler.asyncInstance)
+            .distinctUntilChanged()
+            .throttle(.milliseconds(inputThrottleInMilliseconds), scheduler: MainScheduler.instance)
+            .map { [unowned self] in
+                self.viewModel.inputs.verifyUserInput(userInput: $0!)
+            }
+        
+        usernameValid
+            .skip(1)
+            .subscribe(onNext: { [weak self] isValid in
+                guard let self = `self` else { return }
+                if isValid {
+                    self.viewModel.inputs.verifyUsernameAvailability(userInput: self.usernameField.text!)
+                } else {
+                    self.usernameMessageLabel.textColor = .red
+                    self.usernameMessageLabel.text = Constants.PubStrings.Warnings.usernameInvalidMessage
+                }
+            })
+            .disposed(by: disposeBag)
+
+        let passwordValid = passwordField
+            .rx
+            .text
+            .observe(on: MainScheduler.asyncInstance)
+            .distinctUntilChanged()
+            .throttle(.milliseconds(inputThrottleInMilliseconds), scheduler: MainScheduler.instance)
+            .map { [unowned self] in
+                self.viewModel.inputs.verifyUserInput(userInput: $0!)
+            }
+        
+        passwordValid
+            .skip(1)
+            .subscribe(onNext: { [weak self] isValid in
+                guard let self = `self` else { return }
+                if isValid {
+                    self.passwordMessageLabel.textColor = .blue
+                    self.passwordMessageLabel.text = Constants.PubStrings.passwordValidMessage
+                } else {
+                    self.passwordMessageLabel.textColor = .red
+                    self.passwordMessageLabel.text = Constants.PubStrings.Warnings.passwordInvalidMessage
+                }
+            })
+            .disposed(by: disposeBag)
+
+        let confirmPasswordValid = confirmPasswordField
+            .rx
+            .text
+            .observe(on: MainScheduler.asyncInstance)
+            .throttle(.milliseconds(inputThrottleInMilliseconds), scheduler: MainScheduler.instance)
+            .map { [unowned self] in
```

Comment:
unnecessary [unowned self] in ?

Reply:
added a variable to hold values that will be evaluated.